keras/layers/core.py

```diff
@@ -33,6 +46,25 @@ def __init__(self, **kwargs):
         if not hasattr(self, 'params'):
             self.params = []
 
+    def __call__(self, X):
```

Comment:
@EderSantana , I don't see any new tests in this pull.. Maybe you forgot to commit it?
As for call overload, I'm not sure what it the policy on keras, but I would personally stay away on overloading 'call' in the core library - it proved to be really really hard to undo or change later in my practice. What ends up happening usually in big projects is that objects that are called with '()' are hard to track. Lets say get_input is changed in some non-obvious way so you have to make sure that 'call' is not going to break things. It is practically impossible to find all the calls: normal 'Find all: "functionX"' is not going to work. So there is always a small call somewhere that interacts in a weird way and breaks something else etc. Another case - if you for some reason decide to get rid of call or change arguments, you again will probably loose some of use cases because you just cant find it. This is a non-compiled language-specific issues since in python an invalid object call will happily just sit until someone actually calls it. So I would just rename it to 'apply_to_input' or add optional argument to 'get_output' or smth like that. Call overload is perfectly fine in the client code though - e.g. you need to send a callback to 'fit' function. However I'm not really aware of all the usecases and practices here, so other more experienced guys like @fchollet should know better:)